configuration/config2/CfgValue.hpp

```diff
@@ -0,0 +1,103 @@
+        const std::string &getDefaultValue() const { return m_default; }
+        bool hasDefaultValue() const { return m_defaultSet; }
+        void setReadOnly(bool readOnly) { m_readOnly = readOnly; }
+        bool isReadOnly() const { return m_readOnly; }
+        void setHidden(bool hidden) { m_hidden = hidden; }
+        bool isHidden() const { return m_hidden; }
+        void setDeprecated(bool deprecated) { m_deprecated = deprecated; }
+        bool isDeprecated() const { return m_deprecated; }
+        void setTooltip(const std::string &tooltip) { m_tooltip = tooltip; }
+        const std::string &getTooltip() const { return m_tooltip; }
+        void addModifer(const std::string &mod) { m_modifiers.push_back(mod); }
+        void setModifiers(const std::vector<std::string> &list) { m_modifiers = list;  }
+        const std::vector<std::string> &getModifiers() const { return m_modifiers; }
+        bool hasModifiers() const { return m_modifiers.size() != 0; }
+        void setUniqueValue(bool isUnique) { m_isUnique = isUnique; }
+        bool isUniqueValue() const { return m_isUnique;  }
+        void setUniqueValueSetRef(const std::shared_ptr<CfgValue> &pValue) { m_pUniqueValueSetRefs.push_back(pValue);  }
+        bool isFromUniqueValueSet() const { return !m_pUniqueValueSetRefs.empty(); }
+        std::vector<std::weak_ptr<CfgValue>> getUniqueValueSetRefs() const { return m_pUniqueValueSetRefs;  }  
+        bool isConfigured() const { return m_isConfigured;  }
+        void resetEnvironment();
+        void setMirrorFromPath(const std::string &path) { m_mirrorFromPath = path;  }
+        const std::string &getMirrorFromPath() const { return m_mirrorFromPath;  }
+        bool isMirroredValue() const { return m_mirrorFromPath.length() != 0; }
+        void addMirroredCfgValue(const std::shared_ptr<CfgValue> &pVal) { m_mirrorToCfgValues.push_back(pVal); }
+        void addEnvValue(const std::shared_ptr<EnvValue> &pEnvValue) { m_envValues.push_back(pEnvValue); }
+        std::vector<std::string> getAllEnvValues() const;
+        void mirrorValue(const std::string &oldValue, const std::string &newValue);
+        void setMirroredEnvValues(const std::string &oldValue, const std::string &newValue);
+        //void validate(Status &status, const std::string &id, const std::string &value, const EnvValue *pEnvValue = nullptr) const;
+        void validate(Status &status, const std::string &id, const EnvValue *pEnvValue = nullptr) const;
+        std::vector<AllowedValue> getAllowedValues(const EnvValue *pEnvValue = nullptr) const;
+
+
+    protected:
+
+        std::vector<std::string> getAllKeyRefValues(const EnvValue *pEnvValue) const;
+
+
+    protected:
+
+        std::shared_ptr<CfgType> m_pType;
+        std::vector<std::weak_ptr<EnvValue>> m_envValues;
+        std::vector<std::shared_ptr<CfgValue>> m_mirrorToCfgValues;
+        std::string m_name;
+        std::string m_displayName;
+        std::string m_mirrorFromPath;
```

Comment:
if many of these strings are not always set, may make sense to use a map to save memory.